src/python/pants/backend/helm/util_rules/deployment.py

```diff
@@ -0,0 +1,306 @@
+
+        return [ContainerRef.parse(cast(str, container["image"])) for container in containers]
+
+
+@dataclass(frozen=True)
+class AnalyseDeploymentRequest:
+    field_set: HelmDeploymentFieldSet
+
+
+@dataclass(frozen=True)
+class AnalysedDeployment:
+    container_images: tuple[ContainerRef, ...]
+
+
+class InjectHelmDeploymentDependenciesRequest(InjectDependenciesRequest):
+    inject_for = HelmDeploymentDependenciesField
+
+
+@rule
+async def get_chart_via_deployment(field_set: HelmDeploymentFieldSet) -> HelmChart:
+    explicit_dependencies = await Get(
+        ExplicitlyProvidedDependencies, DependenciesRequest(field_set.dependencies)
+    )
+    explicit_targets = await Get(
+        Targets,
+        Addresses(
+            [
+                addr
+                for addr in explicit_dependencies.includes
+                if addr not in explicit_dependencies.ignores
+            ]
+        ),
+    )
+
+    found_charts = [tgt for tgt in explicit_targets if HelmChartFieldSet.is_applicable(tgt)]
+    if not found_charts:
+        raise MissingHelmDeploymentChartError(
+            f"The target address '{field_set.address}' is missing a dependency on a `helm_chart` target."
+        )
+    if len(found_charts) > 1:
+        raise TooManyChartDependenciesError(
+            f"The target address '{field_set.address}' has too many `helm_chart` addresses in its dependencies, it should have only one."
+        )
+
+    chart_fs = HelmChartFieldSet.create(found_charts[0])
+    return await Get(HelmChart, HelmChartFieldSet, chart_fs)
+
+
+@rule(desc="Analyse Helm deployment dependencies", level=LogLevel.DEBUG)
+async def analyse_deployment_dependencies(request: AnalyseDeploymentRequest) -> AnalysedDeployment:
+    chart, value_files = await MultiGet(
+        Get(HelmChart, HelmDeploymentFieldSet, request.field_set),
+        Get(
+            StrippedSourceFiles,
+            SourceFilesRequest([request.field_set.sources]),
+        ),
+    )
+
+    rendered_chart = await Get(
+        RenderedChart,
+        RenderChartRequest(
+            chart,
+            value_files=value_files.snapshot,
+            values=request.field_set.values.value or FrozenDict[str, str](),
+        ),
+    )
+    file_contents = await Get(DigestContents, Digest, rendered_chart.snapshot.digest)
+
+    manifests = [
+        ResourceManifest.from_dict(parsed_yaml)
+        for file in file_contents
+        for parsed_yaml in yaml.safe_load_all(file.content)
+    ]
+    logger.debug(
+        f"Found {pluralize(len(manifests), 'manifest')} in the chart '{chart.metadata.name}' rendered by deployment at: {request.field_set.address}"
+    )
+    return AnalysedDeployment(
+        container_images=tuple(
+            chain.from_iterable([manifest.container_images for manifest in manifests])
+        )
+    )
+
+
+@rule(desc="Find container images being used by a Helm deployment", level=LogLevel.DEBUG)
+async def inject_deployment_dependencies(
```

Comment:
Awesome!

Reply:
thanks! I has a curious side effect though, since to able to determine the actual repository of a docker image we need an instance of the Docker build context. This causes the Docker backend to collect all sources and dependencies needed to build a Docker image.

Is there a way of just _querying_ what is the repository name for a given docker image without the need of actually building the Docker build context?